### DISCUSSION TITLE
Can we have an option to add additional Keywords f

### DISCUSSION THREAD

**Discussion by sfc-gh-nakincilar** [dawnlabs/carbon#1400] [Ideas]
I am looking to add custom keywords for to be highlighted for SQL to factor for various versions

**Comment by ghost** [comment]
I've created issue #1534 I originally intended to add additional C# tokens but noticed that we could do it for other languages too. I can add SQL to that list. 

If you're looking to just modify original tokens this can already be achieved by selecting text. Once you've selected something a new color picker appears on the right-hand side of the UI allowing you to explicitly set a color for only the selected text. It's admittedly a little slower manually selecting and coloring each time over the automation but that may help you for now as a band-aid patch solution.

![Carbon](https://github.com/user-attachments/assets/65742592-447c-4511-bd0c-3d4ee81ef854)

For disambiguation, when you said custom did you mean something along the lines of adding a text box to type in a word and then have it color all instances of that word? Sort of like a find and replace, only find and colorize. I think that would be a pretty neat feature but would likely take a little time to implement. I might be able to tackle that one though.
